lib/Browser.js

```diff
@@ -183,7 +183,12 @@ class Browser extends EventEmitter {
    * @return {!Promise<!Array<!Puppeteer.Page>>}
    */
   async pages() {
-    const pages = await Promise.all(this.targets().map(target => target.page()));
+    const nonVisiblePageTypes = ['service_worker', 'background_page'];
```

Comment:
I think this can be done simpler as something like
```js
return Promise.all(this.targets().filter(target => target.type === 'page').map(target => target.page())));
```

Reply:
@JoelEinbinder - done, 10x!